validator/tests/test_client_request_handlers/mocks.py

```diff
@@ -110,20 +110,25 @@ def add_block(self, base_id, root=b'merkle_root'.hex()):
 
         signer_public_key = b'public_key' + bytes(base_id, 'utf-8')
 
+        batch = make_mock_batch(base_id)
+
         header = BlockHeader(
             block_num=num,
             previous_block_id=previous_id,
             signer_public_key=signer_public_key.hex(),
-            batch_ids=[block_id],
+            batch_ids=[batch.header_signature],
             consensus=b'consensus',
             state_root_hash=root)
 
         block = Block(
             header=header.SerializeToString(),
             header_signature=block_id,
-            batches=[make_mock_batch(base_id)])
+            batches=[batch])
 
-        self.put_blocks([block])
```

Comment:
Isn't this still required?

Reply:
Just the way the diff shows things. It's still in the `add_block` method, not in the new `make_mock_block` method.